### PR TITLE
Fix GetFirstWalletBlockLocator

### PR DESCRIPTION
**Pull request by quantumagi**
Fixes a `Sequence contains no elements` error which occurs when the data directory contains no wallets.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs [hunk 1]
```diff
@@ -1500,7 +1500,7 @@ public Wallet GetWalletByName(string walletName)
         /// <inheritdoc />
         public ICollection<uint256> GetFirstWalletBlockLocator()
         {
-            return this.Wallets.First().BlockLocator;
+            return this.Wallets.FirstOrDefault()?.BlockLocator ?? new[] { this.network.GenesisHash };
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs]: If there's no wallet, "GetFirstWalletBlockLocator" shouldn't even be called. If it is called and there is no wallet, I think it's better to return null or an empty collection.

**Comment by bokobza** [comment]
I've restarted a build for ubuntu, which you can see here: https://dev.azure.com/StratisProject/StratisBitcoinFullNode/_build/results?buildId=11608

File: src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs]: Unfortunately returning 0 items will cause the calling code to fail:

```
                ICollection<uint256> locators = this.walletManager.GetFirstWalletBlockLocator();
                var blockLocator = new BlockLocator { Blocks = locators.ToList() };
                ChainedHeader fork = this.chain.FindFork(blockLocator);
                this.walletManager.RemoveBlocks(fork);
                this.walletManager.WalletTipHash = fork.HashBlock;
                this.walletTip = fork;
```

`FindFork` would return null.

Comment (bokobza) [reply src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs]: the problem is that if the wallet tip is the genesis, you're going to be syncing from the beginning when you create a wallet.

Comment (bokobza) [reply src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs]: Also, the block locator is located in the wallet file, not sure what you'd do with a block locator if you don't have a wallet.

Comment (quantumagi) [reply src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs]: Perhaps the way to think about this is that the wallet tip keeps track of the blocks we no longer need to deal with in order to keep our current set of wallets up-to-date.

That would imply that the tip can be the current chain tip in the case where there are no wallets.

**Pull request merged**